tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,284 @@
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+          readOnly: true,
+        },
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToData, stepCapPerRun)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        this._canceller.cancelAll();
+        const router = getRouter();
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        const urlRunsPortion = this.runs.map(r => `run=${r}`).join('&');
+        url += `tag=${this.tag}&${urlRunsPortion}`;
+        const updateData = this._canceller.cancellable(result => {
+          if (result.cancelled) {
+            return;
+          }
+
+          const runToData = result.value;
+          this.set('_runToData', runToData);
+          this.fire('data-updated', {
+            'runToData': runToData,
+          });
+        });
+        this.requestManager.request(url).then(updateData);
+      },
+      _setChartData(runToData, stepCapPerRun) {
+        if (!runToData) {
+          return;
+        }
+
+        _.forOwn(runToData, (entries, run) => {
+          if (!entries || !entries.length) {
+            return;
+          }
+
+          const stepCap = stepCapPerRun[run];
+          let entriesIndex = entries.length - 1;
+          while (entriesIndex > 0) {
+            if (entries[entriesIndex].step <= stepCap) {
+              // This is the rightmost entry closest (or equal to) the step cap.
+              break;
+            }
+            entriesIndex--;
+          }
+          const entry = entries[entriesIndex];
+
+          // Update the step shown for this run within the card.
+          this._actualStepPerRun[run] = entry.step;
+          this._actualWallTimePerRun[run] = entry.wall_time;
+
+          // This clone operation is necessary for Polymer to notify an update.
+          this.set('_actualStepPerRun', _.clone(this._actualStepPerRun));
+
+          // Reverse the values so they are plotted in order, which allows for
+          // tool tips.
+          const precision = entry.precision.slice().reverse();
+          const recall = entry.recall.slice().reverse();
+
+          let seriesData = [];
+          for (let i = 0; i < entry.precision.length; i++) {
+            seriesData.push({
+              scalar: precision[i],
```

Comment:
I understand why this needs to be called scalar, but IMO it's pretty ugly (and non-obvious unless you know how `vz-line-chart` works). What do you think about modifying `vz-line-chart` such that it's possible to just pass `{scalar, precision}` data?